packages/medusa/src/services/product.js

```diff
@@ -182,23 +181,21 @@ class ProductService extends BaseService {
 
     const { q, query, relations } = this.prepareListQuery_(selector, config)
 
+    let products
+    let count
     if (q) {
-      const qb = this.getFreeTextQueryBuilder_(productRepo, query, q)
-      const [raw, count] = await qb.getManyAndCount()
-
-      const products = await productRepo.findWithRelations(
+      ;[products, count] = await productRepo.getFreeTextSearchResultsAndCount(
```

Comment:
comment: Never seen syntax like this `;[variable]`. Where did you come across that and what does it do? 😅

Reply:
does not do much, it would have been to separate two steps made on the same line. example:
```ts
doSomething()[0].forEach(...) // We take the first element of the returned array from doSomething
doSomething();[0].forEach(...) // Now they are two separate actions
// With the actual format
doSomething()
;[0].forEach(...)
```

Reply:
[suggestion: Remove the `;`

Reply:
Very short [read](https://prettier.io/docs/en/rationale.html#semicolons) that I think will do a much better job at explaining that I will 😆 

TL;DR: it's added by prettier as a protective measure

Reply:
Yes this is true, never thought that someone would miss it but hey we never know 😂